cal/cal/helpers.py

```diff
@@ -28,6 +31,90 @@ def ensure_timezone_awareness(dt, optional_timezone=None):
     return dt
 
 
+def handle_time_string(time_str, timezone_str):
+    """
+    Converts UTC 'time_str' to timezone represented by 'timezone_str'
+    """
+    timezone = None
+    if timezone_str:
+        try:
+            timezone = pytz.timezone(timezone_str)
+        except pytz.UnknownTimeZoneError:
+            raise Exception("{} could not be parsed into a timezone".format(timezone_str))
+
+    time = parse_datetime(time_str)
+    if not time:
+        # Parse the date and create a datetime at the zeroth hour
+        date = parse_date(time_str)
+        if not date:
+            raise Exception("{} couldn't be parsed as date or datetime".format(time_str))
+        time = datetime.combine(date, datetime.min.time())
+
+    if timezone_util.is_naive(time):
+        if timezone:
+            time = timezone_util.make_aware(time, timezone)
+        else:
+            time = timezone_util.make_aware(time, timezone_util.get_default_timezone())
+
+    if timezone:
+        time = time.astimezone(timezone)
+    else:
+        time = time.astimezone(timezone_util.utc)
+
+    return time
+
+
+def get_time_series(model, timezone='UTC', time_range='Weekly', start=None, end=None):
+    """
+    Returns a list of week-hour tuples corresponding to the events in this ColorCategory.
+    Each week starts at the start time.
+    """
+    week_hours = []
+    events = model.query().order_by('start')
+    i = 0
+    # Convert start to local time
+    start = handle_time_string(str(events[0].start), timezone)
+    if time_range == 'Daily':
+        # To indicate do nothing if Daily is passed in
+        pass
+    if time_range == 'Weekly':
+        # Change start date to be Monday beginning of week
+        while start.weekday() != 0:
+            start = start - timedelta(days=1)
+    if time_range == 'Monthly':
+        # Change start date to beginning of month
+        while start.day != 1:
+            start = start - timedelta(days=1)
```

Comment:
@cindytung Here's the updated generalized function that can take in 'Daily', 'Weekly', "Monthly' and will do everything accordingly :) all you need to do is call this function with different arguments in your next PR.